Casket additions and adjustments

Omnione opened the pull request:
• Moved loot tables to casket_loot.lua as it was a bloating caskets.lua.
• Added all remaining zone mixins, treasure_casket.lua's and checked zone ID text tables.
• Added all remaining loot tables for zones added.
• Fixed an issue where a hint would return an incorrect number.
• Fixed an issue where you cannot enter the number 10 as an input.

<!-- place 'x' mark between square [X] brackets to affirm: -->
**_I affirm:_**
- [X] that I agree to Project Topaz's [Limited Contributor License Agreement](http://project-topaz.com/blob/release/CONTRIBUTOR_AGREEMENT.md), as written on this date
- [X] that I've _tested my code_ since the last commit in the PR, and will test after any later commits

Omnione committed:
Casket additions and adjustments

• Moved loot tables to casket_loot.lua as it was a bloating caskets.lua.
• Added all remaining zone mixins, treasure_casket.lua's and checked zone ID text tables.
• Added all remaining loot tables for zones added.

Omnione committed:
Casket adjustment

• Added a check in timeElapsedCheck to catch an npc being nil.
• removed a stray i = i +1 in getCasketID

ibm2431 commented on scripts/globals/caskets.lua:
```diff
@@ -243,8 +98,10 @@ local function timeElapsedCheck(npc)
 
-    spawnTime = npc:getLocalVar("[caskets]SPAWNTIME")
-
+    if npc ~= nil then
+        spawnTime = npc:getLocalVar("[caskets]SPAWNTIME")
+    end
+    
```
Thoughts on doing:
```lua
if npc == nil then
    return false
end
```
To immediately abort the rest of the function?

Omnione replied:
thats a good call i'll do that

Omnione replied:
im gonna add a check the see if the casket has a local var set for its spawn time too.

Omnione replied:
Added the the suggested change plus a check for the local var.

ibm2431 commented on scripts/globals/casket_loot.lua:
```diff
@@ -0,0 +1,4462 @@
+            { 390, 12987}, -- Ebony Sabots
+        }
+    },
+        [tpz.zone.QUICKSAND_CAVES] =
```
Got an extra indent on this key

Omnione replied:
stupid copy pasta

Omnione replied:
Fixed

Omnione committed:
Casket adjustments

• Added check to see if the npc is nil in timeElapsedCheck, if it is, it will return flase and about the spawn.
• removed extra indent in casket_loot.lua at line 1047.

ibm2431 reviewed (approved):
Thanks, Omni!
